src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/TypeLiteralNode.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Syntax.SourceInformation;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    // CallNode :: Type ( TypeLiteralNode )
+    // CallNode :: IsType ( TypeLiteralNode )
+    // CallNode :: AsType ( Expr, TypeLiteralNode )
+
```

Comment:
Can you expand this a bit, detail why we need a TypeLiteralNode as it's own thing?